### DISCUSSION TITLE
can't get heap/bin command working correctly while run gdbserver with remote android

### DISCUSSION THREAD

**Discussion by yzcmf** [pwndbg/pwndbg#1898] [General]
No matter using gef, pwndbg or pwngdb or peda; While runing those commands in android phone with gdbserver, never get the heap command work ? Is there anything with libc.so ? Should I install libc.so debug version in android ? I don't know how to install the debug version of libc.so in android phone ?  
Anyone know how to solve this problem

gef➤  heap arenas
gef➤  heapinfo
Can not get libc version
Cannot get main_arena's symbol address. Make sure you install libc debug file (libc6-dbg & libc6-dbg:i386 for debian package).
Can't find heap info
gef➤  heapbase
Python Exception <class 'FileNotFoundError'>: [Errno 2] No such file or directory: '/proc/12607/maps'
Python Exception <class 'gdb.error'>: Error occurred in Python: [Errno 2] No such file or directory: '/proc/12607/maps'
Error occurred in Python: Error occurred in Python: [Errno 2] No such file or directory: '/proc/12607/maps'
gef➤  parseheap
Can not get libc version
Cannot get main_arena's symbol address. Make sure you install libc debug file (libc6-dbg & libc6-dbg:i386 for debian package).
can't find heap info
gef➤  heap
[!] Syntax
heap (chunk|chunks|bins|arenas|set-arena)
gef➤  heap chunks
[!] Invalid arena
gef➤

**Comment by disconnect3d** [comment]
Hi, Android uses bionic libc, not glibc which we "understand". I don't know if bionic runs the same ptmalloc2 implementation as glibc, but if not, that's why there is the discrepancy and issues.

I would be very happy to support other libc debugging, but I don't have time to work on it myself. If you want it implemented, please either do it and send us a PR, or, we could think of some paid project to do that.
